docs/reference/how-to/search-speed.asciidoc

```diff
@@ -424,3 +424,147 @@ The <<text,`text`>> field has an <<index-phrases,`index_prefixes`>> option that
 indexes prefixes of all terms and is automatically leveraged by query parsers to
 run prefix queries. If your use-case involves running lots of prefix queries,
 this can speed up queries significantly.
+
+[[faster-filtering-with-constant-keyword]]
+=== Use <<constant-keyword,`constant_keyword`>> to speed up filtering
```

Comment:
A suggestion for restructuring this section a bit: it could start with something like "Imagine that you have an index containing cycles. There are a large number of bicycles, and many searches perform a filter on `cycle_type: bicycle`." Then you could explain how to speed up the searches by splitting the index in two and switching to `constant_keyword`. The point about `match_all` getting rewritten away could be saved for a later paragraph as detail about how the optimization works.

This approach seems nice to me because it immediately introduces the set-up and suggested optimization (which is intuitive even without knowing the details of query rewriting). Leading with the `match_all` justification might make it harder to understand for people who aren't as familiar with the search DSL.